Jenkinsfile

```diff
@@ -9,63 +9,87 @@ if (env.BRANCH_IS_PRIMARY) {
+          "GPG_FILE=${WORKSPACE}/credentials/sandbox.gpg",
+          "GPG_KEYNAME=Bogus Test",
+          "GPG_PASSPHRASE=s3cr3t",
+          "GPG_PASSPHRASE_FILE=${WORKSPACE}/credentials/test.gpg.password.txt",
+          "HOME=/home/jenkins/agent/workspace",
+          "WAR=${WORKSPACE}/jenkins.war",
+          "MSI=${WORKSPACE}/jenkins.msi",
+          "RELEASELINE=-experimental",
+        ]) {
+          stage('Preparation') {
+            checkout scm
+            sh './prep.sh'
+          }
+
+          stage('Build') {
+            sh 'make package && python3 -m pytest bin --junitxml target/junit.xml'
+            junit 'target/junit.xml'
+            def results = '*.war, target/debian/*.deb, target/rpm/*.rpm, target/suse/*.rpm'
+            stash includes: results, name: 'results'
+            archiveArtifacts results
+          }
+        }
       }
+    }
 
-      stage('Build') {
-        sh 'make package && python3 -m pytest bin --junitxml target/junit.xml'
-        junit 'target/junit.xml'
-        def results = '*.war, target/debian/*.deb, target/rpm/*.rpm, target/suse/*.rpm'
-        stash includes: results, name: 'results'
-        archiveArtifacts results
+    nodeWithTimeout('docker') {
+      stage('Test') {
+        checkout scm
+        unstash 'results'
+        infra.withDockerCredentials {
+          ansiColor('xterm') {
+            sh '''
+                cat /proc/cpuinfo
+                cat /proc/meminfo
+                python3 -m venv venv
+                . venv/bin/activate
+                pip install -U pip wheel
+                pip install -r requirements.txt
+                ANSIBLE_FORCE_COLOR=true molecule test
+                ANSIBLE_FORCE_COLOR=true molecule test -s servlet
+                deactivate
+            '''.stripIndent()
+          }
+        }
       }
     }
-  }
-}
+  }, Windows: {
+    nodeWithTimeout('docker-windows') {
+      stage('Preparation') {
+        checkout scm
+      }
 
-nodeWithTimeout('docker') {
-  stage('Test') {
-    checkout scm
-    unstash 'results'
-    infra.withDockerCredentials {
-      ansiColor('xterm') {
-        sh '''
-            cat /proc/cpuinfo
-            cat /proc/meminfo
-            python3 -m venv venv
-            . venv/bin/activate
-            pip install -U pip wheel
-            pip install -r requirements.txt
-            ANSIBLE_FORCE_COLOR=true molecule test
-            ANSIBLE_FORCE_COLOR=true molecule test -s servlet
-            deactivate
-        '''.stripIndent()
+      stage('Build') {
+        withEnv([
+          "GPG_FILE=${WORKSPACE}/credentials/sandbox.gpg",
+          "GPG_KEYNAME=Bogus Test",
+          "GPG_PASSPHRASE=s3cr3t",
+          "GPG_PASSPHRASE_FILE=${WORKSPACE}/credentials/test.gpg.password.txt",
+          "WAR=${WORKSPACE}/jenkins.war",
+          "MSI=${WORKSPACE}/jenkins.msi",
+          "RELEASELINE=-experimental",
+        ]) {
+          powershell ".\\make.ps1"
```

Comment:
So while this works its not the same image as run in the windows build currently.

https://github.com/jenkins-infra/release/blob/master/Jenkinsfile.d/core/package#L259

Reply:
Is there a way to use the same pod template from the release repo? Is the pod template for the Linux build the same as used in release?

Reply:
Good grief @timja . However, given how slow the Windows pod are to spin up (the dotnet SDK container takes 12 to 17 min to pull and AKS does not allow nodes to come with container images preloaded AFAIK), I would not mind using a Windows (2022?) VM with the same template as in ci.jio.

@slide does the current `docker-windows` work? if yes, we can totally set up release to use the same.

The only important part is the timing: we need to sync the PR change here with the release agent definition change to avoid breaking next release

Reply:
`docker-windows` is using the packer image setup, is that correct? Also, I am not sure if it works, I don't have access to test my Jenkinsfile changes on CI as I am not part of the packaging repo team.

Reply:
> `docker-windows` is using the packer image setup, is that correct? 

That is correct!

Reply:
> Also, I am not sure if it works, I don't have access to test my Jenkinsfile changes on CI as I am not part of the packaging repo team.

Is is it ok and worth it to ask for you to have maintainer permission on this?

Reply:
(failed: https://ci.jenkins.io/job/Packaging/job/packaging/job/PR-496/2/pipeline-console/?selected-node=51)

Reply:
Do I just request access via helpdesk?

Reply:
> Do I just request access via helpdesk?

Ideally yes: it is easier to audit afterwards. We tend to also ask in jenkins dev mailing list to have +1 (to gather community feedbacks) as an unspoken but usual rule, to get a quorum.

That should be a quick one

Reply:
I sent an email to the developers list and got some +1's: https://groups.google.com/g/jenkinsci-dev/c/U5WoCd77wuo

Reply:
added permission, I think you will need to create a new PR for the Jenkinsfile change to work